webflow-hacks/misc-hacks/6/6.ts

```diff
@@ -0,0 +1,69 @@
+document.addEventListener('DOMContentLoaded', function () {
+  const WAIT_TIME = 4500;
+  const COOKIE_NAME = 'seenGif';
+
+  const LOADING_WRAPPER_SELECTOR = '[fs-hacks-element="loading-wrapper"]';
+  const GIF_SELECTOR = '[fs-hacks-element="gif"]';
+  const CLEAR_BUTTON_SELECTOR = '[fs-hacks-element="clear-cookie"]';
+  const loadingWrapper = document.querySelector<HTMLDivElement>(LOADING_WRAPPER_SELECTOR);
+  const clearButton = document.querySelector<HTMLDivElement>(CLEAR_BUTTON_SELECTOR);
+  if (!loadingWrapper || !clearButton) return;
+
+  const seenGif = getCokie(COOKIE_NAME);
+  if (seenGif) return;
+
+  loadingWrapper.style.display = 'flex';
+
+  playGif(GIF_SELECTOR);
+  setCokie(COOKIE_NAME, 'true', 1);
+
+  setTimeout(async () => {
+    loadingWrapper.style.display = 'none';
+  }, WAIT_TIME);
+
+  clearButton.addEventListener('click', function () {
+    removeCookie(COOKIE_NAME);
+  });
+});
+
+/**
+ * define the playGif function
+ * @param {string} gifclass
+ **/
+function playGif(gifclass: string) {
+  // get the gif's class and store it in a variable
+  const gif = document.querySelector<HTMLImageElement>(gifclass);
```

Comment:
I don't think `giffclass` is the best name for this particular use case, as it can be any type of selector.
Also that comment is quite redundant, I think we can get rid of it!